File: dairy_erp/dairy_erp/page/dairy_dashboard/dairy_dashboard.js
var i = 0;
frappe.pages['dairy-dashboard'].on_page_load = function(wrapper) {
    new frappe.dairy_dashboard({
        $wrapper: $(wrapper)
    });
    frappe.breadcrumbs.add("Dairy Erp");
}

frappe.pages['dairy-dashboard'].refresh = function(wrapper) {
    if (i!=0) {
        location.reload()
    }
    i=i+1
}

frappe.dairy_dashboard = Class.extend({
    init : function(opts){
        this.$wrapper = opts.$wrapper
        this.render_layout();
    },

    render_layout: function() {
        this.$wrapper.empty();
        this.$wrapper.append(frappe.render_template("dashboard_layout", {"header": "Dairy"}));
        this.$sidebar = this.$wrapper.find("#dairy_sidebar");
        this.$content = this.$wrapper.find(".dairy_content");
        this.$total_row = this.$wrapper.find('.total_summery');
        this.make();
    },

    make: function() {
        this.make_sidebar();
        this.fetch_data();
        this.bind_event();
    },

    make_sidebar: function() {
        side_menus = [{
            "label": "Head Office",
            "doctype": "Address",
            "add_type": "Head Office"
        },
        {
            "label": __("Camp Office"),
            "doctype": "Address",
            "add_type": "Camp Office"
        },
        {
            "label": "Chilling Center",
            "doctype": "Address",
            "add_type": "Chilling Centre"
        },
        {
            "label": "Plant Office",
            "doctype": "Address",
            "add_type": "Plant"
        },
        {
            "label": "VLCC",
            "doctype": "Village Level Collection Centre",
            "add_type": ""
        },
        {
            "label": "Supplier",
            "doctype": "Supplier",
            "add_type": ""
        },
        {
            "label": "Material Price List",
            "doctype": "Material Price List",
            "add_type": ""
        }]

        this.$sidebar.append(frappe.render_template('dashboard_sidebar', {'side_menus': side_menus}))
    },

    fetch_data: function() {
        var me = this;
        frappe.call({
            method:"dairy_erp.dairy_erp.page.dairy_dashboard.dairy_dashboard.get_data",
            callback : function(r){
                me.data = r.message
                me.render_view();
                me.bind_event()
            }
        })
    },

    render_view: function() {
        var me = this;
        me.$content.append(frappe.render_template("dairy_dashboard",{"data":me.data || {}}))
        me.$total_row.append(frappe.render_template('total_summery', {
            'data': me.data['total_summery']
        }))
    },

    bind_event : function(){
        var me = this;
        if(me.data){
            $.each(me.data.addr,function(i,d){
                if(d && d.address_type == "Head Office"){
                    $('#head-office').hide()
                }
            })
        }
        $("#head-office").on("click",function(){
            frappe.route_options = {
                "address_type": "Head Office"
            };
            frappe.new_doc("Address")
        })
        $("#camp-office").on("click",function(){
            frappe.route_options = {
                "address_type": "Camp Office",
                "Dynamic Link.link_doctype" : "Company",
                "Dynamic Link.link_name" : "Dairy"
            };
            frappe.new_doc("Address")
        })
        $("#chilling-centre").on("click",function(){
            frappe.route_options = {
                "address_type": "Chilling Centre"
            };
            frappe.new_doc("Address")
        })
        $("#plant").on("click",function(){
            frappe.route_options = {
                "address_type": "Plant"
            };
            frappe.new_doc("Address")
        })
        $("#new_vlcc").on("click",function(){
            frappe.new_doc("Village Level Collection Centre")
        })
        $("#new_supp").on("click",function(){
            frappe.route_options = {
                "supplier_type": "Dairy Local"
            };
            frappe.new_doc("Supplier")
        })

        $(".camp-office").on("click",function(){
            frappe.set_route("List", "Address", {'address_type': "Camp Office"});
        })
        $(".head-office-list").on("click",function(){
            frappe.set_route("List", "Address", {'address_type': "Head Office"});
        })
        $(".chilling-centre").on("click",function(){
            frappe.set_route("List", "Address", {'address_type': "Chilling Centre"});
        })
        $(".plant").on("click",function(){
            frappe.set_route("List", "Address", {'address_type': "Plant"});
        })

        $('._doctype').on("click", function() {
            doctype = $(this).attr('data-doctype');
            add_type = $(this).attr('data-add_type');
            filters =  add_type ? {"address_type": add_type} : {}
            frappe.set_route("List", doctype, filters)
        })
        $(".supplier-list").on("click",function(){
            frappe.set_route("List", "Supplier", {'supplier_type': "Dairy Local"});
        })

        // Refresh Btn
        $('.refresh_btn').on("click", function() {
            window.location.reload();
        })
    }
})

File: dairy_erp/dairy_erp/page/dairy_dashboard/dairy_dashboard.py
from __future__ import unicode_literals
import frappe
from frappe.model.document import Document
from dairy_erp.dairy_erp.page.vlcc_dashboard.vlcc_dashboard import get_total_counts


@frappe.whitelist()
def get_data():
    address = frappe.db.sql("""select name,address_type,pincode,address_line1,state,address_line2,city,country,email_id,phone,fax 
                                from `tabAddress`""",as_dict=1)
    camp_off = 0
    plant = 0 
    cc = 0
    if address:
        for addr in address:
            if addr.get('address_type') == 'Head Office':
                office_addr = address_manipulation(addr)
                addr.update({"office_addr":office_addr})
                final_addr = ""
            else:
                if addr.get('address_type') == "Camp Office":
                    if camp_off != 2:
                        office_addr = address_manipulation(addr)
                        addr.update({"office_addr":office_addr})
                        final_addr = ""
                        camp_off += 1
                if addr.get('address_type') == "Plant":
                    if plant != 2:
                        office_addr = address_manipulation(addr)
                        addr.update({"office_addr":office_addr})
                        final_addr = ""
                        plant += 1
                if addr.get('address_type') == "Chilling Centre":
                    if cc != 2:
                        office_addr = address_manipulation(addr)
                        addr.update({"office_addr":office_addr})
                        final_addr = ""
                        cc += 1
        vlcc = get_vlcc_data()
        supplier = get_supplier_data()
        vlcc_name = frappe.get_value("User", frappe.session.user, "company")
        return {"addr":address,"vlcc":vlcc,"supplier":supplier, "total_summery": get_total_counts("Vlcc Milk Collection Record")}

def address_manipulation(addr):
    final_addr = ""
    if addr.get('address_line1'):
        final_addr += addr.get('address_line1') + "<br>"
    if addr.get('address_line2'):
        final_addr += addr.get('address_line2') + "<br>"
    if addr.get('city'):
        final_addr += addr.get('city') + "<br>" 
    if addr.get('state'):
        final_addr += addr.get('state') + "<br>" 
    if addr.get('pincode'):
        final_addr += addr.get('pincode') + "<br>" 
    if addr.get('country'):
        final_addr += addr.get('country') + "<br>" 
    if addr.get('phone'):
        final_addr += addr.get('phone') + "<br>"
    if addr.get('fax'):
        final_addr += addr.get('fax') + "<br>"
    if addr.get('email_id'):
        final_addr += addr.get('email_id') + "<br>"
    return final_addr

def get_vlcc_data():
    return frappe.db.sql("""select name,address_display as addr from `tabVillage Level Collection Centre` limit 2""",as_dict=1)

def get_supplier_data():
    address = frappe.db.sql("""select a.name,a.address_type,a.pincode,a.address_line1,a.state,a.address_line2,a.city,a.country,a.email_id,a.phone,a.fax, s.name as supplier,s.supplier_type 
                               from `tabSupplier` s  
                               left join `tabDynamic Link` d on d.link_name = s.name 
                               left join `tabAddress` a on d.parent = a.name 
                               where s.supplier_type = 'Dairy Local' limit 2""",as_dict=1)
    if address:
        for addr in address:
            office_addr = address_manipulation(addr)
            addr.update({"addr":office_addr})
            final_addr = ""
        return address